Resize slideshow dialog

dstopp opened the pull request:
- Fixes #1059 
- Were any Django, import pipeline, table_schema, or dictionary files modified? Y/N
  - If YES:
    - Database used: opus3_test_200903
    - All Django tests pass: Y
    - FLAKE8 run on modified code: NA
- Were any JavaScript or CSS files modified? Y
  - If YES:
    - JSHINT run on all affected files: Y
    - Tested on Chrome, Firefox, Safari, Edge, and iOS: Y
      (Remember to test all browser sizes)
    - Tested for race conditions (with delayed API calls if applicable): NA

Description of changes:
-Allow the galleryView/slide show modal to be resized;
-Provide min/max icons for instant resize;
-move the image to the top of the column when the modal narrows;
-change the font size and icon fonts to smaller when the modal becomes small;
-minor changes to update class names for code consistancy;
-Docking code is present but currently commented out

Known problems:

dstopp committed:
removed array.prototype and replaced w/diff code due to prototype behaviour that joined into array

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into prototype_array_bug

dstopp committed:
-on menu click, check if search widget has constraints.  If so, pop up confirm modal.  On confirm, delete search widget OR cancel based on confirm modal selection

dstopp committed:
-minor title update
-verify if checkbox type, only do checkbox check.  If input:text type, only check val()

dstopp committed:
-bug fix where the left menu did not get updated when Surface Geometry Target was closed and there were sub-widgets selected for that target that were empty.
-fixed logic for checking empty widget

dstopp committed:
-remove default from closeCard and closeAndRemoveWidgetFromDOM
-minor grammer corrections in comments
-remove spaces

dstopp committed:
-misc update of class names to code standards;
-remove unused file
-add .resizable to slide

dstopp committed:
-fix the height direction for resize

dstopp committed:
-make the gallery/table loader have a greater z-index so it is on top of the modal, and remove the modal-specific loader.
-shrink the metadata menu font size at the same time the slide is shrunk on resize so they have the same font size.
-fix the direction the menu pops up <or down> based on what item is selected.

dstopp committed:
-minor changes working on the responsive behaviour of the image

dstopp committed:
added icons for min/max

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

juzen2003 committed:
Make gallery modal resize veritcally & horizontally without resizing the
image proportionally.

dstopp committed:
-update class to code standards
-add docking framework <not done>

dstopp committed:
Merge pull request #13 from juzen2003/1059_resize_dock_slideshow_dialog

Make gallery modal resize vertically & horizontally without resizing the image proportionally

dstopp committed:
zoom/shrink implemented; docking is currently commented out

dstopp committed:
change the minimize icon and upgrade fontawesome <required for new icon>

dstopp committed:
fixes odd spacing when growing and shrinking modal back and forth

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
added comment

juzen2003 commented on opus/application/static_media/css/opus.css:
```diff
@@ -1104,27 +1107,120 @@ img.op-slideshow-image-preview {
+
+.op-tune-image-size {
+    height: 88% !important;
+    /*align-self: center; */
```
Remove unwanted style.

juzen2003 commented on opus/application/static_media/css/opus.css:
```diff
@@ -1104,27 +1107,120 @@ img.op-slideshow-image-preview {
 
 #galleryView .modal-content {
     border: solid 2px #FFF;
+    /*max-height: 80vh !important;*/
```
Remove unwanted style.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/css/opus.css:
```diff
@@ -1104,27 +1107,120 @@ img.op-slideshow-image-preview {
+    height: 10% !important;
+    align-items: center !important;
+}
+/* #################################### */
```
Sorry I forgot to remove this when creating pull to your repo.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -281,7 +296,46 @@ var o_browse = {
         });
 
-        $('#galleryView').on("click", "a.op-cart-toggle", function(e) {
+        $('.op-slide-minimize').on("click", function(e) {
```
Use double quotes to meet our js code styles.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -281,7 +296,46 @@ var o_browse = {
+            });
+        });
+
+        $('.op-slide-maximize').on("click", function(e) {
```
Use double quotes.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -281,7 +296,46 @@ var o_browse = {
+            }
+        });
+
+        $('.op-slide-dock').on("click", function(e) {
```
Use double quotes.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -281,7 +296,46 @@ var o_browse = {
+            $(".op-gallery-view-docked").show();
+        });
+
+        $('.op-gallery-view-body').on("click", "a.op-cart-toggle", function(e) {
```
Use double quotes.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -291,7 +345,7 @@ var o_browse = {
         });
 
-        $('#galleryView').on("click", "a.op-prev,a.op-next", function(e) {
+        $('.op-gallery-view-body').on("click", "a.op-prev,a.op-next", function(e) {
```
Use double quotes.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -1126,7 +1180,74 @@ var o_browse = {
+            $target.addClass("op-resize-small");
+            // need to resize the add metadata menu as well and X in the corner
+            $target.find("i.fa-times-circle").removeClass("fa-lg");
+            $(`#op-add-metadata-fields .op-select-list`).addClass("op-resize-small");
```
Since there is no string interpolation, we can use " instead of `, same at line 1205.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -1126,7 +1180,74 @@ var o_browse = {
+            $target.find("i.fa-times-circle").addClass("fa-lg");
+            $(`#op-add-metadata-fields .op-select-list`).removeClass("op-resize-small");
+        }
+        $("#op-gallery-view-content .row.bottom").removeClass( function(index, css) {
```
An extra space is between ( and function.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -2234,13 +2355,13 @@ var o_browse = {
+        let container = ".op-gallery-view-body .op-metadata-details";
         let containerHeight = $(container).height(calculatedContainerHeight);
-        let browseDialogHeight = $(`#galleryViewContents .op-metadata-details .contents`).height();
+        let browseDialogHeight = $(`.op-gallery-view-body .op-metadata-details .contents`).height();
```
Since there is no string interpolation, we can use " instead of `, same at line 2378, 2379, 2383, 2548, and 2589.

dstopp replied:
done

juzen2003 commented on opus/application/static_media/js/cart.js:
```diff
@@ -645,8 +645,8 @@ var o_cart = {
-        $(`#galleryViewContents .op-cart-toggle`).attr("title", `${buttonInfo[tab].title} (spacebar)`);
-        $(`#galleryViewContents .op-cart-toggle`).html(`<i class="${buttonInfo[tab].icon} fa-2x float-left"></i>`);
+        $(`.op-gallery-view-body .op-cart-toggle`).attr("title", `${buttonInfo[tab].title} (spacebar)`);
+        $(`.op-gallery-view-body .op-cart-toggle`).html(`<i class="${buttonInfo[tab].icon} fa-2x float-left"></i>`);
```
Replace ` with " in the selector.

dstopp replied:
done

juzen2003 reviewed (changes requested):
- Observed issues:
    - When the modal is first open, its size is a bit larger than the default size.
        - Steps to reproduce:
            1. Start at browse tab freshly (like after a refresh), and open up the gallery view modal (default size). 
            2. Click the .op-slide-maximize button, the modal size will shrink a little bit (very little). 
    - The default size of the modal will not change when browser screen is resized, not sure if this is the expected behavior.
       - Steps to reproduce:
            1. Start at browse tab freshly and open up the gallery view modal.
            2. Click the .op-slide-maximize button and close the modal.
            3. Reduce the height of browser window to be smaller than the modal height 
            4. Open the modal again, the modal will be big and cover the whole window. If we don't click the .op-slide-maximize button at step b, the modal will open with the correct size.
    - After a certain amount of modal resizing and moving, the newly open modal will always stay at the right and outside of the browser window.
        - Steps to reproduce:
            1. Start at browse tab freshly.
            2. Open a modal, resize & move, and then click the .op-slide-maximize button and close the modal.
            3. Open a modal again, the modal will stay outside of the browser window (at the right). Repeat step b if the modal open up within the browser window. 
    - When modal height is short and the image is still at the left, the metadata detail text will overlap with the ".op-cart-toggle" icon. Example size: 505 x 292

dstopp committed:
-fix minor code standard adjustments
-remove unnecessary comments

dstopp committed:
hide the galleryView slide modal on resize

dstopp committed:
only close the slide view modal when the browser is resized

dstopp committed:
interim change to fix browser resize issues acting on modal

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

# Conflicts:
#	opus/application/static_media/js/browse.js

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
fixup of bad merge

dstopp committed:
fix modal resizing when the browser is resized.
animate the resizing for max

dstopp committed:
remove unused function

dstopp committed:
changes to center the modal when the browser is resized, slide minimized or slide maximized.

dstopp committed:
make the docking tools no-select

dstopp committed:
-fix centering and  animation for min/max of slide

dstopp committed:
After maximize, make sure that as the browser size changes, the slide will also change responsively.

dstopp committed:
attempt to make the font not 'overshrink' when the browser gets small.  Note there is still a tiny 'jump' at the media break

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
-fixed the spacing issues on the bottom line when the font-size changed;
-fixed the size of the perfect scrollbar container area so there is no overlap

dstopp committed:
-rename 'galleryView' to 'op-metadata-detail-view', as well as update corresponding functions
-added some code for help w/debugging the resizable issues

dstopp committed:
Fix for min/max centering

-fix typo in css;
-add 15% from top for default modal positioning
-add flag in the resizable element to keep track of min/max

dstopp committed:
Update the upper right icons

-make the close button look like the other tools at the bottom of the slide;
-disable min/max if the slide is already at minimized/maximized size

dstopp committed:
More tweaks for centering and font-size

-Add media break for the modal so font size is controlled by code, not media break
-change centering so that it only moves the modal if the modal goes out of the viewable screen
-minor bugs w/max/min icons not quite correct

dstopp committed:
More adjustments, specific to changing browser size

-only make metadata detail view and metadata select modals draggable
-force resize and draggable to stay inside the viewable window
-resize the modal when the browser is resized and keep it in the viewable portion

dstopp committed:
More changes for browser resize

note there are still some console.log messages for debugging; not done.

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

# Conflicts:
#	opus/application/static_media/js/browse.js

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

dstopp committed:
Update code, address comments and suggestions from review

dstopp committed:
Fixed weird recenter/size issue 

Use content instead of dialog for height & width

dstopp committed:
Add icon in gallery view to image indicating last viewed slide

Include title info regarding icon

dstopp committed:
Fix the 'Edit' message for metadata 'drag to reorder'

dstopp committed:
Better align the wrap of 'Drag to reorder'

dstopp committed:
Fixed min size issue, slide show moving out of view and offset 'edit'

juzen2003 commented on opus/application/static_media/js/browse.js:
```diff
@@ -1140,20 +1194,248 @@ var o_browse = {
+        }
+
+        if (top < 0) {
+            top = 0
```
missing `;`

dstopp replied:
done

juzen2003 reviewed (changes requested):
- Database used: opus3_test_210526
- All Django tests pass: Y
- JSHINT run on all affected files: Y
- FLAKE8 run on all affected files: Y
- Tested on Chrome, Firefox, Safari, and iOS: Y
  (Remember to test all browser sizes)
- Tested for race conditions (with delayed API calls if applicable): NA

- Issue:
  - When the slide show dialog is not min/max, the size of the modal will increase every time when the browser window is resize (either increase/decrease the window size).
      - steps to reproduce:
          1. Make the slide show dialog small manually, and keep the browser window relatively big.
          2. Slightly increase the height of the browser window, the modal size will increase
          3. Slightly decrease the height of the browser window, the modal size will keep increasing a little bit.
          4. Keep doing step 2 & 3, we can see that modal size keep increasing but browser window size is not changing.

dstopp committed:
Minor fix for size change and typo

dstopp committed:
Merge branch 'master' of https://github.com/SETI/pds-opus into 1059_resize_dock_slideshow_dialog

# Conflicts:
#	opus/application/static_media/js/browse.js

rfrenchseti reviewed (changes requested):
- When a thumbnail has the mini-binoculars showing (it was the last observation in the slideshow), you can't click anywhere near the top of the thumbnail and have the slideshow come up again. The entire top of the thumbnail is nonresponsive. Thumbnails that don't have the mini-binoculars showing work fine. Note the same thing happens when "DETAIL" is shown. If you click in the area near the DETAIL, nothing happens. I think in all cases the slideshow should appear.
- When you have a browser that is <= 9 thumbnail rows tall, when you click on a thumbnail the default position for the slideshow extends below the footer. It's one thing to allow people to drag the slideshow dialog that low, but something else to make that the default position.
- If you keep clicking on a thumbnail and closing the dialog by hitting ESC without moving the mouse, the dialog will show up in a different vertical position each time.

dstopp committed:
allow modal to be opened on DETAIL and mini-binocular elements

dstopp committed:
Fix the minor size change, location and esc/dialog moving issues
